ledger/participant-integration-api/src/main/scala/platform/apiserver/ratelimiting/RateLimitingInterceptor.scala

```diff
@@ -83,13 +83,15 @@ object RateLimitingInterceptor {
       metrics: Metrics,
       config: RateLimitingConfig,
       additionalChecks: List[LimitResultCheck] = List.empty,
+      indexDbQueueSizeProvider: () => Long,
```

Comment:
We don't need this if we wrap the index DB queue size check in a `LimitResultCheck`

Reply:
How would this work? We need a way to read the current queue size

Reply:
I would do it in SandboxOnXRunner.buildRateLimitingInterceptor